src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -1917,6 +1917,10 @@ void MacroAssembler::ror_imm(Register dst, Register src, uint32_t shift, Registe
 void MacroAssembler::andi(Register Rd, Register Rn, int64_t imm, Register tmp) {
   if (is_imm_in_range(imm, 12, 0)) {
     and_imm12(Rd, Rn, imm);
+  } else if (is_power_of_2(~(imm - 1))) {
```

Comment:
Since this has been reworked, is this change still necessary then? I assume this won't make a difference in number of instructions emitted as compared with the final else branch?